redwood/api/src/tasks/syncStarknetState.ts

```diff
@@ -148,3 +153,20 @@ export const readCids = async (cid: CID | null) => {
     return {videoCid: null, photoCid: null}
   }
 }
+
+export const sendNotifications = async (profile: CachedProfile) => {
+  if (profile.challengeTimestamp !== null) {
+    await maybeNotify(
+      {
+        type: 'NEW_CHALLENGE',
+        profileId: profile.id,
+        challengeTimestamp: profile.challengeTimestamp.toISOString(),
```

Comment:
I like the approach of having a `maybeNotify` — seems like a clever/reusable higher-order function!

[ISO date strings don't seem canonical](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toISOString) so I think probably better to use the actual timestamp integer here.

(Naming-wise, I think we also want to disambiguate between datetime strings and timestamps rather than using) `timestamp` to mean either)

Reply:
(An example of non-canonicality — https://en.wikipedia.org/wiki/ISO_8601#Times)

Reply:
The date strings as serialized by JS are canonical. (The ISO spec gives multiple options for separators, short vs long form, etc, but the ECMA spec for `toISOString` that you linked to is stricter about what implementations are allowed to output.)

Regarding renaming the columns to explicitly reference whether they're a Unix timestamp or a DateTime: In general, I find I use longer variable names and encode more type information in the variable name when using a dynamically-typed language like JS or Ruby than when using a statically-typed language like TS or Kotlin. When using TS the type information is available whenever you hover over a variable, so I don't find I get a lot of value from repeating that information in the variable name itself.